### PR TITLE
GitHub Deployments: support simple repo connect

### PR DESCRIPTION
**Pull request by mpkelly**
Related to https://github.com/Automattic/dotcom-forge/issues/5333

See a summary of changes + video here. p1707809411913779-slack-C06D9M3CHMK

Screenshot 2567-02-12 at 16 20 10

Screenshot 2567-02-12 at 16 22 26

## Proposed Changes

This isn't 100% done but can be reviewed now. 

* Support connecting a simple repo
    - List repos for an account where the app is installed
    - Connect to a repo and specify some basic config
* Supports filtering repos by account 
* Support a local filter of repos by name (filters local results only)

Not done: 
- pagination of results. Discussion p1707728079969959-slack-C06D9M3CHMK. 
- connecting to another branch that isn't the default

## Testing Instructions

* Apply patch D137683-code which adds a new endpoint `GET /wpcom/v2/hosting/github/repositories`
* Also make sure you have the access token hardcoded as per previous setup steps
* Start Calypso on this branch and go to `/github-deployments/:atomic-site`
* Select a repo
* Connect the repo and check the network response is 200

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11334297

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @mpkelly for including a screenshot in the description! This is really helpful for our translators.

File: client/my-sites/github-deployments/connected/repositories/connect-repository.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+	const { createDeployment, isPending } = useCreateCodeDeployment( selectedSiteId, {
+		onSuccess: () => {
+			dispatch( successNotice( __( 'Deployment created.' ), noticeOptions ) );
+		},
+		onError: ( error ) => {
+			dispatch(
+				errorNotice(
+					// translators: "reason" is why connecting the branch failed.
+					sprintf( __( 'Failed create deployment: %(reason)s' ), { reason: error.message } ),
```
Comment (emilyaudela) [inline client/my-sites/github-deployments/connected/repositories/connect-repository.tsx:53]: Should this be "Failed to create deployment"?

Comment (mpkelly) [reply client/my-sites/github-deployments/connected/repositories/connect-repository.tsx:53]: Yes, it should. Thanks, @emilyaudela. I added this patch, https://github.com/Automattic/wp-calypso/pull/87476, which I will merge once the build passes.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11334297).

**Pull request merged**